main.go

```diff
@@ -396,12 +396,26 @@ func getTracer(opts Options, out output) (opentracing.Tracer, io.Closer) {
 	return opentracing.NoopTracer{}, nil
 }
 
+type healthSerializer interface {
+	ThriftHealthSerializer() encoding.Serializer
+	ProtoHealthSerializer(serviceName string) encoding.Serializer
+}
+
 // withTransportSerializer may modify the serializer for the transport used.
 // E.g. Thrift payloads are not enveloped when used with TChannel or gRPC.
-func withTransportSerializer(p transport.Protocol, s encoding.Serializer, rOpts RequestOptions) encoding.Serializer {
+func withTransportSerializer(p transport.Protocol, s encoding.Serializer, opts Options) encoding.Serializer {
+	if opts.ROpts.Health && s.Encoding() == encoding.UnspecifiedEncoding {
```

Comment:
rather than checking for health / encoding, can we just check for the interface?